tests/common/test_event.py

```diff
@@ -270,107 +298,81 @@ def test_periodic_emits_after_elapsed_delta(self, mock_event):
-                elif i['name'] == 'EventPid':
-                    self.assertEqual(i['value'], 42)
-                elif i['name'] == 'TaskName':
-                    self.assertEqual(i['value'], 'HelloWorldTask')
-                elif i['name'] == 'KeywordName':
-                    self.assertEqual(i['value'], '')
-                elif i['name'] == 'GAVersion':
-                    self.assertEqual(i['value'], str(CURRENT_AGENT))
-                else:
-                    self.assertFalse(True, "Contains a field outside the defaults expected. Field Name: {0}".
-                                     format(i['name']))
-
-    def test_save_event_message_with_decode_errors(self):
-        tmp_file = os.path.join(self.tmp_dir, "tmp_file")
-        fileutil.write_file(tmp_file, "This is not JSON data", encoding="utf-16")
-
-        for tld_file in os.listdir(self.tmp_dir):
-            try:
-                MonitorHandler.collect_event(os.path.join(self.tmp_dir, tld_file))
-            except Exception as e:
-                self.assertIsInstance(e, EventError)
+        for filename in os.listdir(self.event_dir):
+            self.assertTrue(filename.endswith(EVENT_FILE_EXTENSION),
+                'Event file does not have the correct extension ({0}): {1}'.format(EVENT_FILE_EXTENSION, filename))
+
+    @staticmethod
+    def _get_event_message(evt):
+        for p in evt.parameters:
+            if p.name == 'Message':
+                return p.value
+        return None
+
+    def test_collect_events_should_be_able_to_process_events_with_non_ascii_characters(self):
+        self._create_test_event_file("custom_script_nonascii_characters.tld")
+
+        event_list = event.collect_events()
+
+        self.assertEquals(len(event_list.events), 1)
+        self.assertEquals(TestEvent._get_event_message(event_list.events[0]), u'World\u05e2\u05d9\u05d5\u05ea \u05d0\u05d7\u05e8\u05d5\u05ea\u0906\u091c')
+
+    def test_collect_events_should_ignore_invalid_event_files(self):
+        self._create_test_event_file("custom_script.tld")
+        self._create_test_event_file("custom_script_utf-16.tld")
+        self._create_test_event_file("custom_script_invalid_json.tld")
+        os.chmod(self._create_test_event_file("custom_script_no_read_access.tld"), 0o200)
+        self._create_test_event_file("custom_script_valid_json.tld")
+
+        with patch("azurelinuxagent.common.event.logger.warn") as mock_warn:
+            event_list = event.collect_events()
+
+            self.assertEquals(len(event_list.events), 2)
+            self.assertTrue(
+                all(TestEvent._get_event_message(evt) == "A test telemetry message." for evt in event_list.events),
+                "The valid events were not found")
+
+            # expected_warn_message = r'Failed to process event file.*custom_script_with_invalid_json.tld.*Error parsing event'
+            # self.assertIsNotNone(re.search(expected_warn_message, str(mock_warn.call_args[0])))
```

Comment:
Remove?

Reply:
will do

Reply:
actually it is an assert i forgot to put back. will do